src/Analyzer.Reports.UnitTests/SarifReportWriterTests.cs

```diff
@@ -138,25 +126,39 @@ private void AssertSarifLog(SarifLog sarifLog, IEnumerable<Types.IEvaluation> ev
                     }
                 }
 
-                run.Results.Count.Should().Be(failedEvaluations.Count);
-
+                var outputResults = new List<List<IResult>>();
                 for (int i = 0; i < failedEvaluations.Count; i++)
                 {
                     var evaluation = failedEvaluations[i];
-                    var result = run.Results[i];
+                    var evalResults = ReportsHelper.GetFailedResultsAsList(evaluation).Distinct().ToList();
+
+                    // dedupe results
+                    if (outputResults.Any(results => results.SequenceEqual(evalResults)))
+                    {
+                        continue;
+                    }
+
+                    var result = run.Results[outputResults.Count];
                     result.RuleId.Should().BeEquivalentTo(evaluation.RuleId);
                     result.Message.Id.Should().BeEquivalentTo("default");
                     result.Level.Should().Be(FailureLevel.Error);
 
-                    var evalResults = new List<Types.IResult>();
-                    TraverseResults(evalResults, evaluation);
-                    result.Locations.Count.Should().Be(evalResults.Count);
+                    if (evalResults.First().SourceLocation.FilePath != templateFilePath.FullName)
+                    {
+                        result.AnalysisTarget.Uri.OriginalString.Should().BeEquivalentTo(templateFilePath.Name);
+                    }
+
                     for (int j = 0; j < evalResults.Count; j++)
                     {
-                        result.Locations[j].PhysicalLocation.ArtifactLocation.Uri.OriginalString.Should().BeEquivalentTo(templateFilePath.Name);
-                        result.Locations[j].PhysicalLocation.Region.StartLine.Should().Be(evalResults[j].LineNumber);
+                        var evalResultFileName = (new FileInfo(evalResults[j].SourceLocation.FilePath)).Name;
```

Comment:
A little cleaner in my opinion:
```suggestion
                        var evalResultFileName = Path.GetFileName(evalResults[j].SourceLocation.FilePath);
```